BedrockServer.cpp

```diff
@@ -1007,12 +1006,48 @@ void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlo
             break;
         }
 
-        // We're about to retry, decrement the retry count.
-        --retry;
+        // If we're shutting down, or have set a specific max retries, we just try several times in a row and then move the command to the blocking queue.
+        int maxRetries = _maxConflictRetries.load();
+        if (maxRetries || _shutdownState.load() != RUNNING) {
+            if (command->processCount > maxRetries) {
+                SINFO("Max retries (" << maxRetries << ") hit in worker, sending '" << command->request.methodLine << "' to blocking queue with size " << _blockingCommandQueue.size());
+                _blockingCommandQueue.push(move(command));
+                return;
+            }
+        } else {
+            // If we're not shutting down, see how long we want to wait until we'll try this command again.
+            size_t millisecondsToWait = 0;
+            switch (command->processCount) {
+                case 1:
+                    millisecondsToWait = 10;
+                    break;
+                case 2:
+                    millisecondsToWait = 25;
+                    break;
+                case 3:
+                    millisecondsToWait = 50;
+                    break;
+                case 4:
+                    millisecondsToWait = 100;
+                    break;
+                case 5:
+                    millisecondsToWait = 250;
+                    break;
+                default:
+                    millisecondsToWait = 500;
+            }
 
-        if (!retry) {
-            SINFO("Max retries hit in worker, sending '" << command->request.methodLine << "' to blocking queue with size " << _blockingCommandQueue.size());
-           _blockingCommandQueue.push(move(command));
+            // Apply jitter. Take a value that's a whole number up to 50% of ideal time. This allows for adding or subtracting up to 25%.
+            millisecondsToWait += ((SRandom::rand64() % (millisecondsToWait / 2)) - (millisecondsToWait / 4));
+            SINFO("Waiting " << millisecondsToWait << "ms before retrying.");
+            if (hasDedicatedThread) {
```

Comment:
Why do we need an if else for this? Why can't we always go in the `else`?

Reply:
What do you mean? the `if` and `else` do different things.

Reply:
I mean why can't we _always_ do `// Otherwise, re-queue and let another thread try again.`? Why sometimes we _need_ to do `// If we have a dedicated socket thread for this command, we can just sleep here.`?